fault-remediation/main.go

```diff
@@ -166,7 +166,17 @@ func setupNonCtrlRuntimeManaged(ctx context.Context, components *initializer.Com
 
 		for event := range components.FaultRemediationReconciler.Watcher.Events() {
 			slog.Info("Event received", "event", event)
-			_, _ = components.FaultRemediationReconciler.Reconcile(gCtx, &event)
+			for i := 1; i <= components.FaultRemediationReconciler.Config.UpdateMaxRetries; i++ {
```

Comment:
so ctrl-runtime pattern is to throw error to retry and then there's built in exponential backoff and all that so updating the "legacy" reconciler to also retry on errors so that the underlying code can rely on throwing errors to trigger retries while still supporting both modes

Reply:
make sense